### PR TITLE
Toggle-collapse in layer selector

### PR DESCRIPTION
**Pull request by mariavkourouni**
[Demo](https://mf-geoadmin3.dev.bgdi.ch/kom_toggle/)

### CODE REVIEW

**Comment by gjn** [comment]
This works great!

@oterral Could you review css?

File: src/components/layermanager/partials/layermanager.html [hunk 1]
```diff
@@ -23,11 +23,12 @@
                 ng-options="(i | gaTimeLabel:layer) for i in layer.timestamps"></select>
       </div>
       <button class="ga-icon ga-btn icon-user"
-              ng-click="showWarning(layer)"></button>
+                ng-click="showWarning(layer)"></button>
```
Comment (oterral) [inline src/components/layermanager/partials/layermanager.html]: Too much space for indent

File: src/components/layermanager/partials/layermanager.html [hunk 2]
```diff
@@ -4,7 +4,7 @@
         'ga-layer-time-enabled': layer.timeEnabled,
         'ga-layer-warning': layer.useThirdPartyData
       }">
-    <div class="ga-layer-infos">
+      <div class="ga-layer-infos">
```
Comment (oterral) [inline src/components/layermanager/partials/layermanager.html]: Remove the 2 spaces

File: src/components/layermanager/partials/layermanager.html [hunk 3]
```diff
@@ -23,11 +23,12 @@
...
       </div>
       <button class="ga-icon ga-btn icon-user"
-              ng-click="showWarning(layer)"></button>
+                ng-click="showWarning(layer)"></button>
       <!--button ng-if="isBodLayer(layer)" class="ga-icon ga-btn icon-info-sign"
-              ng-click="displayLayerMetadata($event, layer)"></button-->
-    </div>
-    <div class="ga-layer-tools">
+                ng-click="displayLayerMetadata($event, layer)"></button-->
+      <button class="ga-icon icon-gear" ng-click="collapsed=!collapsed" href="#subSelection"></button>
```
Comment (oterral) [inline src/components/layermanager/partials/layermanager.html]: you don't need the `onclick` I think, and the `href`attribute doesn't exist for `<button>`.

File: src/components/layermanager/partials/layermanager.html [hunk 4]
```diff
@@ -23,11 +23,12 @@
...
+    </div><!--ga-layer-infos-->
+    <div id="subSelection" class="ga-layer-tools" ng-click="collapsed">
```
Comment (oterral) [inline src/components/layermanager/partials/layermanager.html]: `ìd` attribute must be a unique identifier and here you are in a `ng-repat` directive so this `ìd` will not be unique, so you can remove it.

The `ng-click` is useless. 
Instead of modifying a scope variable with the `ng-click` and rely on it,  I suggest you do all by css.
As you have seen a `.ga-layer-folded` css class is added when the `ga-layer-tools` div element is hidden and this class is removed when it is shown. 

So you just have to replace the appearance effect of the layer tools from right-left to up-down. 
To summarize here are the 2 things to do: 
  - 1st,  hide the `ga-layer-tools` behind the  `ga-layer-infos` element when the `ga-layer-folded` class is set
  - 2nd, replace the right-to-left transition appearance effect of the `ga-layer-tools`element  to an up-to-down transition and increase the size of the `li` in the same time.

**Comment by oterral** [comment]
@mariavkourouni I've made some comments, I'm not far if you have questions :)

File: src/components/layermanager/style/layermanager.less [hunk 5]
```diff
@@ -3,7 +3,6 @@
 
   li {
     border-bottom: #e9e9e9 1px solid;
-    height: 3em;
     padding: 0;
     overflow: hidden;
```
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: use `height: 6em;`

File: src/components/layermanager/style/layermanager.less [hunk 6]
```diff
@@ -128,16 +127,16 @@
   .ga-layer-time-enabled .ga-layer-infos label {
     padding-right: 7.7em;
   }
- 
+
   .ga-layer-tools {
-    position: absolute;
+    position: relative;
```
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: let  `position:absolute`

File: src/components/layermanager/style/layermanager.less [hunk 7]
```diff
@@ -128,16 +127,16 @@
...
     height: 100%;
     width: 100%;
     top: 0;
     left: 0;
     background: #fff;
     border-left: 3px solid #666;
-    .transition(margin-left .35s ease);
+    .transition(margin-top.35s ease);
```
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: apply the transition on `top` property.

File: src/components/layermanager/style/layermanager.less [hunk 8]
```diff
@@ -149,7 +148,7 @@
     }
   }
  
-  .ga-layer-tools, .ga-layer-infos {
+  .ga-layer-infos {
     height: 100%;
```
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: `height: 3em`

File: src/components/layermanager/style/layermanager.less [hunk 9]
```diff
@@ -190,9 +188,11 @@
   }
 
   .ga-layer-folded {
+    height: 3em;
 
     .ga-layer-tools {
-      margin-left: 100%;
+      top: -3em;
```
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: `top:0em`

File: src/components/layermanager/style/layermanager.less (hunk 6 above)
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: and use the `z-index` property to hide it behind the `ga-layer-infos'

File: src/components/layermanager/style/layermanager.less [hunk 10]
```diff
@@ -149,7 +148,7 @@
     }
   }
  
-  .ga-layer-tools, .ga-layer-infos {
```
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: leave `.ga-layer-tools, .ga-layer-infos`

File: src/components/layermanager/style/layermanager.less [hunk 11]
```diff
@@ -190,9 +188,11 @@
   }
 
   .ga-layer-folded {
+    height: 3em;
```
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: you can remove this.

File: src/components/layermanager/style/layermanager.less [hunk 12]
```diff
@@ -128,16 +127,16 @@
...
     padding-right: 7.7em;
   }
- 
+
   .ga-layer-tools {
-    position: absolute;
+    position: relative;
     height: 100%;
     width: 100%;
     top: 0;
```
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: add  `top: 3em;`

File: src/components/layermanager/style/layermanager.less [hunk 13]
```diff
@@ -190,9 +188,11 @@
   }
 
   .ga-layer-folded {
+    height: 3em;
 
     .ga-layer-tools {
-      margin-left: 100%;
+      top: -3em;
+      z-index: -1;
```
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: Ah nice you already specify a z-index here

**Comment by oterral** [comment]
Much better, there is still improvments to use nice css transition.

File: src/components/layermanager/style/layermanager.less (hunk 11 above)
Comment (mariavkourouni) [inline src/components/layermanager/style/layermanager.less]: this is needed, otherwise it becomes 6em

**Comment by mariavkourouni** [comment]
thank you very much for the review!

File: src/components/layermanager/style/layermanager.less [hunk 14]
```diff
@@ -128,16 +128,16 @@
...
-    height: 100%;
+    height: 3em;
     width: 100%;
-    top: 0;
+    top: 3em;
     left: 0;
     background: #fff;
     border-left: 3px solid #666;
-    .transition(margin-left .35s ease);
+    .transition(bottom.35s ease);
```
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: `bottom` transition can't work if you don't modify the bottom property. Use `top` instead.

File: src/components/layermanager/style/layermanager.less (hunk 11 above)
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: yes

File: src/components/layermanager/style/layermanager.less [hunk 15]
```diff
@@ -128,16 +128,16 @@
   .ga-layer-time-enabled .ga-layer-infos label {
     padding-right: 7.7em;
   }
- 
+
   .ga-layer-tools {
     position: absolute;
-    height: 100%;
+    height: 3em;
```
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: Remove it. Already specified l153.

File: src/components/layermanager/style/layermanager.less [hunk 16]
```diff
@@ -190,9 +189,11 @@
   }
 
   .ga-layer-folded {
+    height: 3em;
 
     .ga-layer-tools {
-      margin-left: 100%;
+      top: 0em;
+      z-index: -1;
```
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: This `z-index` doesn't work as expected the element is behind the `<li>`parent. Remove it. And put 2 different z-index to `.ga-layer-tools` and `.ga-layer-infos`

File: src/components/layermanager/style/layermanager.less [hunk 17]
```diff
@@ -150,7 +150,7 @@
   }
  
   .ga-layer-tools, .ga-layer-infos {
-    height: 100%;
+    height: 3em;
     margin-bottom: 0;
```
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: The `margin-bottom` seems useless

File: src/components/layermanager/style/layermanager.less [hunk 18]
```diff
@@ -3,7 +3,7 @@
 
   li {
     border-bottom: #e9e9e9 1px solid;
-    height: 3em;
+    height: 6em;
     padding: 0;
     overflow: hidden;
```
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: add a transition on height: `.transition(heigth .35s ease);`

File: src/components/layermanager/style/layermanager.less [hunk 19]
```diff
@@ -128,16 +130,15 @@
...
     position: absolute;
-    height: 100%;
     width: 100%;
-    top: 0;
+    top: 3em;
     left: 0;
     background: #fff;
     border-left: 3px solid #666;
-    .transition(margin-left .35s ease);
+    .transition(top.35s ease);
```
Comment (oterral) [inline src/components/layermanager/style/layermanager.less]: Put a space between `top` and `.35`

**Comment by oterral** [comment]
Look svery good to me.
Can  you  squash your commits please.

**Comment by oterral** [comment]
Thanks you very much @mariavkourouni !
